### DISCUSSION TITLE
How secure is it having the API key added?

### DISCUSSION THREAD

**Discussion by wifiuk** [alandtse/tesla#140] [General]
So as we know the API key is the key to the car basically?

By adding it into this integration, how secure is it, is there anything that gives access to that key to anyone other than where it is installed, and communication with tesla servers?

**Comment by alandtse** [comment]
Any HA component can access the tokens. It's as secure as HA or your HA server.

**Reply by wifiuk** [reply]
And if the integration is removed what is left over in terms of residual token/cache

**Reply by alandtse** [reply]
Nothing so long as no other components saved the info.
